scaffold-eth/scaffold-eth-2
# fix: typescript eslint + up eslint

## Description

Fixes Typescript-eslint and updates eslint related packages

to test add this to any of the components:
```
const t = 5; // 't' is assigned a value but never used.eslint[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)

type Test = {}; // 'Test' is defined but never used.eslint[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars) , 
// The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.eslint[@typescript-eslint/no-empty-object-type](https://typescript-eslint.io/rules/no-empty-object-type)

// @ts-nocheck // Do not use "@ts-nocheck" because it alters compilation errors.eslint[@typescript-eslint/ban-ts-comment](https://typescript-eslint.io/rules/ban-ts-comment)

const age: any = "seventeen"; // any is allowed, no error
console.log(age);
```

also I removed this rule
```
"prettier/prettier": [
      "warn",
      {
        "endOfLine": "auto"
      }
    ]
```
because looks like it's redundant, also I think default `ls` is better. You can check how it works by removing last line in any of checked files


See also
https://typescript-eslint.io/getting-started/
https://typescript-eslint.io/rules/no-unused-vars#how-to-use
https://typescript-eslint.io/rules/no-explicit-any
https://typescript-eslint.io/rules/ban-ts-comment/
https://github.com/prettier/eslint-plugin-prettier?tab=readme-ov-file#configuration-new-eslintconfigjs


## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

Fixes #901
